src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -134,17 +148,39 @@ private void FlattenConcatArg(BoundExpression lowered, ArrayBuilder<BoundExpress
         /// Determines whether an expression is a known string concat operator (with or without a subsequent ?? ""), and extracts
         /// its args if so.
         /// </summary>
-        /// <param name="loweredCanReturnNull">
-        /// True if this method returns true, and the expression can return null (string.Concat(object) can return null if
-        /// the object's ToString method returns null)
-        /// </param>
         /// <returns>True if this is a call to a known string concat operator, false otherwise</returns>
-        private bool TryExtractStringConcatArgs(BoundExpression lowered, out ImmutableArray<BoundExpression> arguments, out bool loweredCanReturnNull)
+        private bool TryExtractStringConcatArgs(BoundExpression lowered, out ImmutableArray<BoundExpression> arguments)
         {
             switch (lowered.Kind)
             {
                 case BoundKind.Call:
-                    return TryExtractStringConcatArgsFromBoundCall((BoundCall)lowered, out arguments, out loweredCanReturnNull);
+                    var boundCall = (BoundCall)lowered;
+                    var method = boundCall.Method;
+                    if (method.IsStatic && method.ContainingType.SpecialType == SpecialType.System_String)
+                    {
+                        if ((object)method == (object)_compilation.GetSpecialTypeMember(SpecialMember.System_String__ConcatStringString) ||
```

Comment:
In general we prefer to compare symbols using regular `==` instead of reference equality, even though it doesn't matter in this case.

Reply:
Noted. I'll have copied that from somewhere else in that file.